internal/rpc/core/mempool.go

```diff
@@ -64,61 +73,70 @@ func (env *Environment) BroadcastTxCommit(ctx context.Context, tx types.Tx) (*co
 	err := env.Mempool.CheckTx(
```

Comment:
The way we're writing this function looks a whole lot like a regular function call with extra steps. We want to block on the result of its computation, I'm not totally sure I understand why we don't just write this as a function call that returns the ResponseCheckTx :thinking:

Reply:
I think that's a good move eventually, but the tests *really* depend on injecting behavior into the call back, so it's harder than it should be to get out of this pickle.

Reply:
Might be worth adding a TODO to document that this is why we can't easily rip it out (I think we've bumped into this before).